### ISSUE TITLE
Possible regression of #79641 - Scrolling jumps to the bottom regardless of items on M1 iOS Simulator

### ISSUE THREAD

**Issue by mbtodorov** [flutter/flutter#119617]
## Details
- The `google_maps_flutter` plugin defines `'EXCLUDED_ARCHS[sdk=iphonesimulator*]' => 'arm64'` in its podspec file.
- The latest iOS 16 emulator option can't run with Rosetta

the above two are causing performance issues (input is slow to respond, lags, glitches, jank) and cause a funky scroll effect, where scrollviews immediately jump to the bottom when scrolling.

**Target Platform:** iOS
**Target OS version/browser:** 16
**Devices:** any iOS 16 emulator

## Related issues
A similar problem is described:
- here https://github.com/flutter/flutter/issues/112325
- and here https://github.com/flutter/flutter/issues/79641

**Comment by exaby73** [comment]
Hello @mbtodorov. Thanks for raising this issue. I can indeed reproduce this issue, but I do not think this is just related to google maps. This might be a regression of https://github.com/flutter/flutter/issues/79641 since I can reproduce the "scroll jump" behavior with a simple `ListView.builder`. 

cc: @jmagman for insights

<details>
<summary>Code Sample including google maps implementation and ListView (both are buggy)</summary>

```dart
import 'package:flutter/material.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      // home: HomePage(),
      home: SimpleListView(),
    );
  }
}

class SimpleListView extends StatelessWidget {
  const SimpleListView({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: ListView.builder(
        itemCount: 100,
        itemBuilder: (context, index) {
          return ListTile(
            title: Text('Item ${index + 1}'),
          );
        },
      ),
    );
  }
}

class HomePage extends StatelessWidget {
  const HomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Google Maps demo')),
      body: const Center(
        child: GoogleMap(
          initialCameraPosition: CameraPosition(
            target: LatLng(37.3967317, -122.1000127),
            zoom: 15,
          ),
        ),
      ),
    );
  }
}
```

</details>

<details>
<summary>Google Maps video</summary>

https://user-images.githubusercontent.com/50555895/215983376-70836f65-367a-44b8-a247-71b239858fbf.mov

</details>

<details>
<summary>ListView video</summary>

https://user-images.githubusercontent.com/50555895/215983527-7337dd01-8f57-4d17-b233-f981911ee0e8.mov

</details>

<details><summary><code>flutter doctor -v</code> (Master)</summary>

```
[!] Flutter (Channel master, 3.8.0-1.0.pre.4, on macOS 13.1 22C65 darwin-arm64, locale en-US)
    • Flutter version 3.8.0-1.0.pre.4 on channel master at /Users/nabeelparkar/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.18.6/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nabeelparkar/fvm/versions/master. Consider adding /Users/nabeelparkar/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a752c2f154 (35 minutes ago), 2023-02-01 00:41:16 -0500
    • Engine revision 1703a39662
    • Dart version 3.0.0 (build 3.0.0-187.0.dev)
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nabeelparkar/Library/Android/sdk/
    • Platform android-33, build-tools 33.0.0
    • ANDROID_SDK_ROOT = /Users/nabeelparkar/Library/Android/sdk/
    • Java binary at: /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave Browser

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[!] Android Studio (version 2022.1)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9477386/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.1 22C65 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Brave Browser 109.1.47.186

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```

</details>

**Comment by exaby73** [comment]
Upon further investigation, this seems like a duplicate of https://github.com/flutter/flutter/issues/112244. This would be fixed by https://github.com/flutter/flutter/issues/94491. Thank you
